specs/lightclient/beacon-chain.md

```diff
@@ -22,36 +24,66 @@
 - [Helper functions](#helper-functions)
   - [`Predicates`](#predicates)
     - [`eth2_fast_aggregate_verify`](#eth2_fast_aggregate_verify)
+  - [Misc](#misc-2)
+    - [`flags_and_numerators`](#flags_and_numerators)
   - [Beacon state accessors](#beacon-state-accessors)
     - [`get_sync_committee_indices`](#get_sync_committee_indices)
     - [`get_sync_committee`](#get_sync_committee)
+    - [`get_base_reward`](#get_base_reward)
+    - [`get_unslashed_participating_indices`](#get_unslashed_participating_indices)
+    - [`get_flag_deltas`](#get_flag_deltas)
+    - [New `get_inactivity_penalty_deltas`](#new-get_inactivity_penalty_deltas)
   - [Block processing](#block-processing)
+    - [New `process_attestation`](#new-process_attestation)
+    - [New `process_deposit`](#new-process_deposit)
     - [Sync committee processing](#sync-committee-processing)
   - [Epoch processing](#epoch-processing)
-    - [Components of attestation deltas](#components-of-attestation-deltas)
+    - [New `process_justification_and_finalization`](#new-process_justification_and_finalization)
+    - [New `process_rewards_and_penalties`](#new-process_rewards_and_penalties)
     - [Final updates](#final-updates)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 <!-- /TOC -->
 
 ## Introduction
 
-This is a standalone beacon chain patch adding light client support via sync committees.
+This is a patch implementing the first hard fork to the beacon chain, tentatively named HF1 pending a permanent name. It has three main features:
+
+* Light client support via sync committees
+* Incentive accounting reforms, reducing spec complexity
+  and [TODO] reducing the cost of processing chains that have very little or zero participation for a long span of epochs
+* Fork choice rule changes to address weaknesses recently discovered in the existing fork choice
 
 ## Constants
 
+### Participation flags
+
 | Name | Value |
-| - | - | 
-| `BASE_REWARDS_PER_EPOCH` | `uint64(5)` |
+| - | - |
+| `TIMELY_HEAD_FLAG` | `0` |
+| `TIMELY_SOURCE_FLAG` | `1` |
+| `TIMELY_TARGET_FLAG` | `2` |
 
-## Configuration
+### Participation rewards
+
+| Name | Value |
+| - | - |
+| `TIMELY_HEAD_NUMERATOR` | `12` |
+| `TIMELY_SOURCE_NUMERATOR` | `12` |
+| `TIMELY_TARGET_NUMERATOR` | `32` |
+| `REWARD_DENOMINATOR` | `64` |
 
-### Constants
+The reward fractions add up to 7/8, leaving the remaining 1/8 for proposer rewards and other future micro-rewards.
+
+### Misc
 
 | Name | Value |
 | - | - |
+| `PARTICIPATION_FLAGS_LENGTH` | `8` |
```

Comment:
why does the flags vector have length `8`?

it seems that we are only using 3 bits from the 8 and if the thinking is that we will eventually use all 8, i don't see why we wouldn't just introduce more bits later in the relevant fork (which we would need to give new meaning to the current unassigned bits anyway...)

Reply:
See the other discussion about uint8 vs bitvector. Uint8 would give us nice packing behavior in the state merkle tree (32 uint8 entries per leaf), but the current `bitvector[8]` results in one bitvector per 32 byte leaf. So this length might as well be `bitvector[256]` if it comes to merkleization efficiency (ignoring serialization here). It should just be refactored to uint8 IMO. Having more than 3 bits per bitvector is nice though, since it will make this part of the state forward-compatible (so with uint8, everyone has 5 unused zero bits until we start using them for new features) at little cost.